Upgraded to Spring Boot 2.2

Buzzardo opened the pull request:
I upgraded to Spring Boot 2.2.0, which necessitated changing the tests to use JUnit 5, replacing the Gradle wrappers, and modifying README.adoc (because of a change in build.gradle that is described in the document).

Commit:
Upgraded to Spring Boot 2.2

I upgraded to Spring Boot 2.2.0, which necessitated changing the tests to use JUnit 5, replacing the Gradle wrappers, and modifying README.adoc (because of a change in build.gradle that is described in the document).

dsyer commented on README.adoc:
```diff
@@ -312,6 +312,8 @@ For Gradle, add the following dependency:
 ====
 [source,java]
 ----
+implementation files('../library/target/multi-module-library-0.0.1-SNAPSHOT.jar')
```
This looks bad to me (fragile). Why did we switch it from `implementation project(':library')`?

Buzzardo replied:
The `implementation project(':library')` method wasn't building for me. I did some reading. Apparently, the Gradle folks prefer the `files` method.

dsyer replied:
Wut!? That's nuts isn't it? At least the way we did it here (it means if you change the version or root name you have to change all the dependent builds). We can ask someone who knows more about Gradle for a second opinion. Cc: @wilkinsona .

Buzzardo replied:
I got it to work. If i run the build from within the application project, it fails on that `library` line. If I run the build from the parent directory, it works. The instructions in the README say to run it from the parent directory, so OK. I want them to work in isolation, too, but so it goes.

Buzzardo replied:
I set it back to what it was.

wilkinsona replied:
FWIW, I've never seen a recommendation to use `files`. You should use a `project` dependency as it allows Gradle to determine the relationship between the projects and the resulting graph of tasks that need to be executed. If you want to build a task in an individual project you should be able to do that from the root directory by specifying the full path of the task. Gradle will then automatically figure out what, if any, other tasks need to be run first.

dsyer commented on complete/application/build.gradle:
```diff
@@ -15,6 +15,12 @@ repositories {
 	implementation 'org.springframework.boot:spring-boot-starter-web'
-	implementation project(':library')
-	testImplementation 'org.springframework.boot:spring-boot-starter-test'
+	implementation files('../library/target/multi-module-library-0.0.1-SNAPSHOT.jar')
```
See above.

Buzzardo replied:
I set it back to what it was.

dsyer commented on complete/application/src/test/java/com/example/multimodule/application/DemoApplicationTest.java:
```diff
@@ -1,16 +1,15 @@
 package com.example.multimodule.application;
 
-import static org.assertj.core.api.Assertions.assertThat;
```
I thought we preferred AssertJ? Why replace this?

Buzzardo replied:
It wasn't working. When I looked up how to replace it, I found the Hamcrest stuff. If you can show me how to get AssertJ to work with JUnit 5, I'd be fine with changing it.

dsyer replied:
It should just work. The Boot team use AssertJ, so you could look at some of their tests I guess. E.g. https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot/src/test/java/org/springframework/boot/builder/SpringApplicationBuilderTests.java

Buzzardo replied:
I got AssertJ to work. Thanks, Dave.

Commit:
Back to AssertJ

Dave Syer said we use AssertJ, so I got AssertJ to work. Thanks, Dave.

Commit:
Gradle build fix

Per advice from Dave Syer, I put the Gradle build for the complete/application project back to using implementation project(':library'). Thanks, Dave.

Commit:
Fixed the README

Per an observation from Dave Syer, I set the implementation files... line back to implementation project(':library'). Thanks, Dave.

dsyer commented on README.adoc:
```diff
@@ -312,6 +312,8 @@ For Gradle, add the following dependency:
 ----
+implementation project(':library')
+----
 compile project(':library')
```
This line should be removed?

Buzzardo replied:
Yup. Thanks.

Buzzardo replied:
Fixed in yet another commit.

dsyer commented on complete/application/src/test/java/com/example/multimodule/application/DemoApplicationTest.java:
```diff
@@ -19,7 +17,7 @@
     @Test
     public void contextLoads() {
-        assertThat(myService).isNotNull();
+        assertThat(myService.message() != null);
```
Should be `assertThat(myService.message()).isNotNull())`?

Buzzardo replied:
I tried that (naturally, since that was the original). I get a no such method error for `isNotNull()`.

dsyer replied:
I think you have the wrong import then. Works for me. Maybe your IDE is fubar?

Buzzardo replied:
I'm not using an IDE. I'm using a text editor and running the build files from the command line. I'll try STS.

Buzzardo replied:
Same thing in STS.

Buzzardo replied:
I got it to work, finally, so now it is back to `isNotNull()`.

Commit:
Removed extraneous line from README.adoc

Dave Syer caught a copy-paste error that I made. I fixed it. Thanks, Dave.

Commit:
Back to isNotNull

I got isNotNull to work, so I went back to that, on Dave Syer's advice. Thanks, Dave.

wilkinsona commented on complete/library/src/test/java/com/example/multimodule/service/MyServiceTest.java:
```diff
@@ -1,15 +1,14 @@
 package com.example.multimodule.service;
 
-import static org.assertj.core.api.Assertions.*;
```
Can we stick with AssertJ please?

wilkinsona commented on complete/library/src/test/java/com/example/multimodule/service/MyServiceTest.java:
```diff
@@ -18,7 +17,7 @@
     @Test
     public void contextLoads() {
-        assertThat(myService.message()).isNotNull();
+        assertThat(myService.message() != null, is(true));
```
Ideally this would be reverted and we'd stick to AssertJ. If we move to Hamcrest this doesn't look idiomatic to me. It should be something like `assertThat(myService.message(), is(notNullValue()));`

dsyer replied:
Looks like the one on the app was reverted to assertj but this one wasn't yet.

Buzzardo replied:
I missed that one. Sorry. Thanks for catching it. Fixed now.

Commit:
Back to AssertJ for another test

I missed the test in the library application. I have corrected that oversight.

Buzzardo commented:
Thanks for all the help, Dave and Andy. :)
